File: 项目/OC项目/百思不得姐(oc)/百思不得姐(oc)/Classes/Me(我)/Setting(设置)/Model/SettingItem.m
//
//  SettingItem.m
//  百思不得姐(oc)
//
//  Created by 金亮齐 on 2017/5/16.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//

#import "SettingItem.h"

@implementation SettingItem
//实现类方法，进行赋值
+ (instancetype)itemWithTitle:(NSString *)title
{
    SettingItem *item = [[self alloc] init];
    
    item.title = title;
    
    return item;
}

@end

File: 项目/Swift项目/斗鱼(Swift)/斗鱼(Swift)/Classes/Profile/设置/Model/SettingItem.swift
//
//  SettingModel.swift
//  斗鱼(Swift)
//
//  Created by 金亮齐 on 2017/6/3.
//  Copyright © 2017年 醉看红尘这场梦. All rights reserved.
//
import UIKit

class SettingItem: NSObject {
    var title:String = ""
    var subTitle:String = ""
    var icon:String = ""
    
    // 保存每一行cell做的事情
    var operationBlock: ((_ indexPath: IndexPath) -> ())?
    
    init(title: String, subTitle: String = "", icon: String = "") {
        self.icon = icon
        self.title = title
        self.subTitle = subTitle
    }
    
    init(dict: [String: Any]) {
        super.init()
        
        setValuesForKeys(dict)
    }
    
    override func setValue(_ value: Any?, forUndefinedKey key: String) { }
}
